nethttp/server.go

```diff
@@ -21,6 +21,7 @@ func (w *statusCodeTracker) WriteHeader(status int) {
 
 type mwOptions struct {
 	opNameFunc    func(r *http.Request) string
+	spanObserver  func(span opentracing.Span, r *http.Request)
```

Comment:
How about this spanObserver?

Reply:
lgtm. Can you add a test that verifies the ability to override the default span observer?

Reply:
add 4 test cases :)